WooCommerce/Classes/Authentication/AuthenticationManager.swift

```diff
@@ -379,6 +379,18 @@ extension AuthenticationManager: WordPressAuthenticatorDelegate {
         }
     }
 
+    /// Presents the Support Interface
+    ///
+    /// - Parameters:
+    ///     - from: UIViewController instance from which to present the support interface
+    ///     - screen: A case from `CustomHelpCenterContent.Screen` enum. This represents authentication related screens from WCiOS.
+    ///
+    func presentSupport(from sourceViewController: UIViewController, screen: CustomHelpCenterContent.Screen) {
```

Comment:
nit: now that there are multiple variations of calls to `presentSupport(from:customHelpCenterContent:)`, how about making `presentSupport(from:customHelpCenterContent:)` the function in the protocol and moving the ones that call this function to the protocol's extension? if there's another variation in the future, we don't have to change the interface and every implementation every time.

something like:

```swift
protocol Authentication {
    func presentSupport(from sourceViewController: UIViewController,
                                        customHelpCenterContent: CustomHelpCenterContent?)
    ...
}

extension Authentication {
    func presentSupport(from sourceViewController: UIViewController, screen: CustomHelpCenterContent.Screen) {
        let customHelpCenterContent = ...
        presentSupport(from: sourceViewController, customHelpCenterContent: customHelpCenterContent) 
    }

    func presentSupport(from sourceViewController: UIViewController,
                        sourceTag: WordPressSupportSourceTag,
                        lastStep: AuthenticatorAnalyticsTracker.Step,
                        lastFlow: AuthenticatorAnalyticsTracker.Flow) {
        let customHelpCenterContent = ...
        presentSupport(from: sourceViewController, customHelpCenterContent: customHelpCenterContent) 
    }
}

Reply:
I don't fully understand this. But, I will share some more details down below. 

1. The method to present support using sourceTag, step and flow is from [the WordPressAuthenticator library](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/trunk/WordPressAuthenticator/Authenticator/WordPressAuthenticatorDelegateProtocol.swift#L71).  So, I don't think adding that to an extension to `Authentication` protocol will make sense.
2. I plan to remove [the present using sourceTag method](https://github.com/woocommerce/woocommerce-ios/blob/feat/7547-jetpack-error-help/WooCommerce/Classes/ServiceLocator/Authentication.swift#L19) from [Authentication protocol](https://github.com/woocommerce/woocommerce-ios/blob/feat/7547-jetpack-error-help/WooCommerce/Classes/ServiceLocator/Authentication.swift) and replace those calls with [custom help center method](https://github.com/woocommerce/woocommerce-ios/blob/feat/7547-jetpack-error-help/WooCommerce/Classes/ServiceLocator/Authentication.swift#L15). I will delete [this method](https://github.com/woocommerce/woocommerce-ios/blob/feat/7547-jetpack-error-help/WooCommerce/Classes/ServiceLocator/Authentication.swift#L19) once I implement custom help centers for `RoleErrorViewController` and `StorePickerViewController`. i.e. Using `sourceTag` inside WCiOS screens will be removed, as the passed in parameter `sourceTag` is not used anywhere.

Kindly let me know if you think this sounds good enough.

Reply:
> 1. The method to present support using sourceTag, step and flow is from [the WordPressAuthenticator library](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/trunk/WordPressAuthenticator/Authenticator/WordPressAuthenticatorDelegateProtocol.swift#L71).  So, I don't think adding that to an extension to `Authentication` protocol will make sense.

ohh I wasn't aware that `presentSupport(from:sourceTag:lastStep:lastFlow:)` is from `WordPressAuthenticatorDelegate` and not `Authentication` protocol 😅 Please ignore the suggestion then.

The plan for the future changes sounds good, feel free to merge the PR then 🚀